### PR TITLE
man: various improvements

### PR DESCRIPTION
**Pull request by UniqMartin**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes?
- [x] Have you successfully ran `brew tests` with your changes locally?

----

Summary:

- Refactor `brew man` into several smaller and more manageable methods (and fix code style).
- Pipe generated Markdown markup directly to `ronn` to generate man page (and its HTML version), making it possible to remove `brew.1.md`. – Addresses https://github.com/Homebrew/brew/pull/10#issuecomment-208849202.
- Generate man page from an ERB template (with brief editing instructions inside) instead of separate header and footer Markdown fragments. – Addresses https://github.com/Homebrew/brew/pull/10#issuecomment-207395306.

*Note:* I think this PR is best reviewed commit by commit. The refactoring makes the full diff less useful.

### CODE REVIEW

File: Library/Homebrew/manpages/brew.1.md.erb [hunk 1]
```diff
@@ -1,4 +1,54 @@
...
+    Perform a substring search of formula names for <text>. If <text> is
+    surrounded with slashes, then it is interpreted as a regular expression.
+    The search for <text> is extended online to some popular taps.
+    If no search term is given, all locally available formulae are listed.
+
+## COMMANDS
+
+<%= commands.join("\n") %>
 
   * `--cache`:
```
Comment (xu-cheng) [inline Library/Homebrew/manpages/brew.1.md.erb]: Should the help messages in the below all be relocated to their corresponding command file?

**Comment by xu-cheng** [comment]
:+1:

File: Library/Homebrew/manpages/brew.1.md.erb (hunk 1 above)
Comment (UniqMartin) [inline Library/Homebrew/manpages/brew.1.md.erb]: Yes. I plan to do this in a follow-up PR once this is merged. In the follow-up PR I'll also have to tweak the sort order (`uses` needs to come before `--cache`).

**Comment by mikemcquaid** [comment]
👍

**Pull request closed without merging**